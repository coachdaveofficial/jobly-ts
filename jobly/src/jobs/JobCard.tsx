import React, { useContext } from 'react'
import { Job } from '../types/types'
import UserContext from '../context/UserContext'
import JoblyApi from '../api';
import JobAppsContext from '../context/JobAppsContext';


export default function JobCard({ id, title, salary, equity, companyHandle }: Job) {
  const {currentUser} = useContext(UserContext);
  const {appliedJobsIds, setAppliedJobsIds} = useContext(JobAppsContext)
  async function handleApply() {
    try {
      await JoblyApi.applyToJob(currentUser?.username, id)
      
      type issues below
      
      setAppliedJobsIds(prevData => {
        console.log(prevData);
        if (prevData !== undefined && prevData !== null) {
          return new Set([...prevData, id]);
        }
        return [id];
      });

      return { success: true }
    } catch (error) {
      console.error("signup failed", error)
      return { success: false, error }
    }
  }
  return (
    <div className="JobCard card my-2" id={`job-${id}`}>
      <div className="card-body">
        <h6 className="card-title">{title}</h6>
        <p>{companyHandle}</p>
        {salary && <div><small>Salary: {'$' + salary.toLocaleString()}</small></div>}
        {equity !== undefined && <div><small>Equity: {equity}</small></div>}

        {
          appliedJobsIds?.find(appliedJobId => appliedJobId = id)
            ?
            <button className='float-end btn btn-primary' disabled>Applied</button>
            :
            <button className='float-end btn btn-primary' onClick={handleApply} >Apply</button>
        }

      </div>

    </div>
  )
}
