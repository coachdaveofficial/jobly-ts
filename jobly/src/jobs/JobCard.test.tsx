import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import JobCard from "./JobCard";

it("matches the snapshot", () => {
    const {asFragment} = render(
        <MemoryRouter>
            <JobCard id={0} title={""} salary={null} equity={null} companyHandle={""} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", () => {
    const {getByText} = render(<JobCard id={0} title={"testJob"} salary={null} equity={null} companyHandle={""} />);

    const applyBtn = getByText("Apply");

    expect(getByText("testJob")).toBeInTheDocument();
    expect(applyBtn).toBeInTheDocument();

});