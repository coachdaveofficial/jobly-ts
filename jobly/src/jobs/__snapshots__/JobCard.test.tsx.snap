// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard card my-2"
    id="job-0"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      />
      <p />
      <div>
        <small>
          Equity: 
        </small>
      </div>
      <button
        class="btn btn-info font-weight-bold text-uppercase float-end"
      >
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
